Moderation provider docs (#2619)
ably/docs: src/pages/docs/chat/moderation/direct/tisane.mdx

---
title: Tisane
meta_description: "Detect and remove unwanted content in a Chat Room using Tisane AI."
---

[Tisane](https://tisane.ai/) is a powerful Natural Language Understanding (NLU) platform that can be used to moderate content in chat rooms.

> The Tisane rule is a rule applied to chat rooms in Ably Chat which enables you to use Tisane's text moderation capabilities to detect and handle inappropriate content before it is published to other users.

Reviewer: ```suggestion
The Tisane integration can be applied to chat rooms so that you can use Tisane's text moderation capabilities to detect and handle inappropriate content before it's published to other users.
```

## Integration setup <a id="setup"/>

> Configure the integration rule in your [Ably dashboard](https://ably.com/accounts/any/apps/any/integrations) or using the [Control API](/docs/account/control-api).

Reviewer: ```suggestion
Configure the integration in your [Ably dashboard](https://ably.com/accounts/any/apps/any/integrations) or using the [Control API](/docs/account/control-api).
```

The following are the fields specific to Tisane configuration:

| Field | Description |
| ----- | ----------- |
| Tisane API key | The API key for your Tisane account. |
| Thresholds | A map of [text moderation categories](https://docs.tisane.ai/apis/tisane-api-response-guide#supported-types) to severity levels (`low`, `medium`, `high`, `extreme`). When moderating text, any message deemed to be at or above a specified threshold will be rejected and not published to the chat room. |
| Default Language | The language to use for content analysis. |
| Model URL (optional) | A custom URL if using a custom moderation model. |

For additional configuration options shared across all before-publish moderation rules, see the [common configuration fields](/docs/chat/moderation#common-config).

## Handling rejections <a id="rejections"/>

> If a message fails moderation and the rule policy is to reject, then it will be rejected by the server.

Reviewer: Same question for Bodyguard, but this suggests that you set up a policy as to what happens when a message fails moderation. But to me it looks like we reject if it's at, or above the threshold, and this isn't any additional config to do something other than reject.

Author: yeah good point, this was misleading. have updated to be more correct + explicit
